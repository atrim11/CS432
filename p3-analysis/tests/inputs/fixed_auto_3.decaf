def int divide(int a, int b) {
    if (b == 0) {
        return -1;  // Avoid divide by zero
    }
    return a / b;
}

def int calculateSum(int x, int y, int z) {
    return x + y + z;
}

int arr[5];
def int main() {
    int result;
    int x;
    int a;
    // Divide by zero edge case
    result = divide(10, 0);  // Should handle divide by zero

    // Parameter mismatch
    result = divide(10, 10);  // Error: divide expects 2 parameters, provided 1

    // Type mismatch in function call
    result = divide(-1, 10);  // Error: first parameter should be int

    // Function call with too many parameters
    result = calculateSum(1, 2, 4);  // Error: Too many parameters provided

    // Variable shadowing
    a = 5;
    a = 20;

    // Array usage
    arr[10] = 100;  // Error: index out of bounds

    x = arr[3] * 2;  // Valid array usage
    x = arr[-1];  // Error: negative index
    // Invalid recursive call
    result = divide(divide(10, 3), 2);  // Error: nested function call with invalid types
    return 12;
}
