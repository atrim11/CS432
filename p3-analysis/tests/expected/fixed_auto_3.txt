Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 divide : (int, int) -> int
 calculateSum : (int, int, int) -> int
 main : () -> int
 arr : int [5]

  FuncDecl name="divide" return_type=int parameters={a:int,b:int} [line 1]
  SYM TABLE:
   a : int
   b : int

    Block [line 1]
    SYM TABLE:

        Block [line 2]
        SYM TABLE:

  FuncDecl name="calculateSum" return_type=int parameters={x:int,y:int,z:int} [line 8]
  SYM TABLE:
   x : int
   y : int
   z : int

    Block [line 8]
    SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 13]
  SYM TABLE:

    Block [line 13]
    SYM TABLE:
     result : int
     x : int
     a : int

